File: webapi/Models/Budget.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace webapi.Models
{
    /// <summary>
    /// Representa um orçamento associado a um utilizador e suas janelas.
    /// </summary>
    [Table("Budgets")]
    public class Budget : CrudModel
    {
        //[Required]
        public int? BudgetNumber { get; set; } = null;

        /// <summary>
        /// O identificador do utilizador associado ao orçamento.
        /// </summary>
        [ForeignKey("User")/*, Required*/]
        public Guid? UserId { get; set; }

        /// <summary>
        /// O identificador do cliente associado ao orçamento.
        /// </summary>
        [ForeignKey("Client")/*, Required*/]
        public Guid? ClientId { get; set; }

        /// <summary>
        /// O utilizador associado ao orçamento.
        /// </summary>
        [JsonIgnore]
        public virtual User? User { get; set; }

        /// <summary>
        /// Cliente - destinatario do orçamento
        /// </summary>
        [DeleteBehavior(DeleteBehavior.NoAction), JsonIgnore]
        public virtual Client? Client { get; set; }

        /// <summary>
        /// Lista de janelas associadas ao orçamento.
        /// </summary>
        [DeleteBehavior(DeleteBehavior.Cascade)]
        public virtual List<BudgetWindow> BudgetWindows { get; set; } = new List<BudgetWindow>();
    }
}

File: vueapp/src/models/budget.js
import CrudModel from './crudModel.js';
import BudgetWindow from './budgetWindow.js';

import {unref} from 'vue'

class Budget extends CrudModel {
    constructor(json = {}) {
        super(json);
        this.userId = json.userId || null;
        this.budgetNumber = json.budgetNumber || null;
        this.budgetWindows = json.budgetWindows ? json.budgetWindows.map(bw => new BudgetWindow(bw)) : null;
        this.clientId = json.clientId || null;
    }

    toFormData() {
        const formData = super.toFormData();
        formData.append('userId', this.getSafeKeyValue(this.userId));
        formData.append('budgetNumber', this.budgetNumber);
        formData.append('budgetWindows', JSON.stringify(this.budgetWindows || []));
        formData.append('clientId', this.getSafeKeyValue(this.clientId));
        return formData;
    }

    toJSON()
    {
        var data = super.toJSON()

        return {
            ...data,
            userId: this.userId,
            budgetNumber: this.budgetNumber,
            budgetWindows: unref(this.budgetWindows),
            clientId: this.clientId
        }
    }
}

export default Budget;